src/grid/grid_collocate_replay.h

```diff
@@ -8,6 +8,12 @@
 
 #include <stdbool.h>
 
+#if defined(NDEBUG)
+# define GRID_COLLOCATE_ASSERT(EXPR) (void)(EXPR)
+#else
+# define GRID_COLLOCATE_ASSERT(EXPR) assert(EXPR)
+#endif
```

Comment:
I'm confused. Isn't this already the [standard behavior](https://en.cppreference.com/w/c/error/assert) of `assert` in C?

Reply:
Ah, now I realize the difference. Your definition actual executes the expression regardless.

Reply:
No, `assert` completely elides the given expression when `NDEBUG` is defined.

Reply:
( This PR is somewhat similar to issue #922 )

Reply:
> 
> 
> Ah, now I realize the difference. Your definition actual executes the expression regardless.

Yes, the `(void)` cast in front is just to suppress warnings about unused result (comparison expression).